internal/compiler/parser/type.rs

```diff
@@ -74,7 +74,7 @@ pub fn parse_struct_declaration(p: &mut impl Parser) -> bool {
         let mut p = p.start_node(SyntaxKind::DeclaredIdentifier);
         p.expect(SyntaxKind::Identifier);
     }
-    p.expect(SyntaxKind::ColonEqual);
+    p.test(SyntaxKind::ColonEqual);
```

Comment:
Perhaps this relaxation would be worth a syntax test? (I understand the updater kind of covers this though, so maybe a direct coverage is not needed)